modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/IgniteSqlFunctions.java

```diff
@@ -361,7 +360,7 @@ public static BigDecimal toBigDecimal(Number value, int precision, int scale) {
      * (see {@link IgniteSqlOperatorTable#DECIMAL_DIVIDE}, their values are ignored at runtime.
      */
     public static BigDecimal decimalDivide(BigDecimal sum, BigDecimal cnt, int p, int s) {
```

Comment:
the same question, if precision doesn't used why we have it here? I propose to get rid it

Reply:
You can not do this without losing inference of precision/scale (See `DECIMAL_DIVIDE` operator in `IgniteSqlOperatorTable`). Since you can not have different numbers of parameters at the validation phase and in runtime.